### PR TITLE
ControlNetHeader structs for versioning

### PR DESCRIPTION
**Pull request by jessemapel**

### CODE REVIEW

**Comment by jessemapel** [comment]
[Create ControlNetHeaderv#### classes](https://trello.com/c/lFfXR4Q1/89-create-controlnetheaderv-classes)

**Comment by jessemapel** [comment]
This may need documentation on the structs in order to make doxygen happy.

File: isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.cpp [hunk 1]
```diff
@@ -43,177 +43,276 @@ namespace Isis {
...
-  QSharedPointer<ControlPoint> ControlNetVersioner::createPointFromV0007(const ControlPointV0007 point){
+  QSharedPointer<ControlPoint> ControlNetVersioner::createPointFromV0007(const ControlPointV0007 point) {
 
   }
 
 
-  void ControlNetVersioner::createHeaderFromV0001(const ControlNetHeaderV0001){
+  /**
+   * Create the internal header from a V0001 header. This will step the header
+   * version up until it reaches the latest version and then internalize it.
```
Comment (kberryUSGS) [inline isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.cpp]: s/internalize/initialize ?

File: isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.cpp [hunk 2]
```diff
@@ -43,177 +43,276 @@ namespace Isis {
...
 
+    createHeaderFromV0002(newHeader);
   }
 
 
-  void ControlNetVersioner::createHeaderFromV0002(const ControlNetHeaderV0002){
-
+  /**
+   * Create the internal header from a V0002 header. This will step the header
+   * version up until it reaches the latest version and then internalize it.
```
Comment (kberryUSGS) [inline isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.cpp]: s/internalize/initialize ?

File: isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.cpp [hunk 3]
```diff
@@ -43,177 +43,276 @@ namespace Isis {
...
+
+    createHeaderFromV0003(newHeader);
   }
 
 
-  void ControlNetVersioner::createHeaderFromV0003(const ControlNetHeaderV0003){
-
+  /**
+   * Create the internal header from a V0003 header. This will step the header
+   * version up until it reaches the latest version and then internalize it.
```
Comment (kberryUSGS) [inline isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.cpp]: s/internalize/initialize ?

File: isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.cpp [hunk 4]
```diff
@@ -43,177 +43,276 @@ namespace Isis {
...
+
+    createHeaderFromV0004(newHeader);
   }
 
 
-  void ControlNetVersioner::createHeaderFromV0004(const ControlNetHeaderV0004){
-
+  /**
+   * Create the internal header from a V0004 header. This will step the header
+   * version up until it reaches the latest version and then internalize it.
```
Comment (kberryUSGS) [inline isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.cpp]: s/internalize/initialize ?

File: isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.cpp [hunk 5]
```diff
@@ -43,177 +43,276 @@ namespace Isis {
...
+
+    createHeaderFromV0005(newHeader);
   }
 
 
-  void ControlNetVersioner::createHeaderFromV0005(const ControlNetHeaderV0005){
-
+  /**
+   * Create the internal header from a V0005 header. This will step the header
+   * version up until it reaches the latest version and then internalize it.
```
Comment (kberryUSGS) [inline isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.cpp]: s/internalize/initialize ?

File: isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.cpp [hunk 6]
```diff
@@ -43,177 +43,276 @@ namespace Isis {
...
+
+    createHeaderFromV0006(newHeader);
   }
 
 
-  void ControlNetVersioner::createHeaderFromV0006(const ControlNetHeaderV0006){
-
+  /**
+   * Create the internal header from a V0006 header. This will step the header
+   * version up until it reaches the latest version and then internalize it.
```
Comment (kberryUSGS) [inline isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.cpp]: s/internalize/initialize ?

File: isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.cpp [hunk 7]
```diff
@@ -43,177 +43,276 @@ namespace Isis {
...
+
+    createHeaderFromV0007(newHeader);
   }
 
 
-  void ControlNetVersioner::createHeaderFromV0007(const ControlNetHeaderV0007){
-
+  /**
+   * Create the internal header from a V0007 header. This will step the header
+   * version up until it reaches the latest version and then internalize it.
```
Comment (kberryUSGS) [inline isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.cpp]: s/internalize/initialize ?

**Comment by jessemapel** [comment]
@kberryUSGS I actually did mean internalize because the last step is to store the header in the m_header field

**Comment by kberryUSGS** [comment]
:+1: Okay, cool!

**Pull request merged**